code/missionui/missionshipchoice.cpp

```diff
@@ -2181,49 +2218,22 @@ void draw_wing_block(int wb_num, int hot_slot, int selected_slot, int class_sele
 			icon = NULL;
 		}
 
-		mask = ~WING_SLOT_LOCKED; 
-		mask &= ~WING_SLOT_IS_PLAYER;
-		if (ship_selection) {
-			mask &= ~WING_SLOT_WEAPONS_DISABLED;
-		} 
-		else {
-			mask &= ~WING_SLOT_SHIPS_DISABLED;
-		} 
-
-		switch(ws->status & mask ) {
-			case WING_SLOT_FILLED:
-			case WING_SLOT_FILLED|WING_SLOT_IS_PLAYER:
-
-				Assert(icon);
+		if (ws->status == SlotStatus::Filled && !(!ship_selection && ws->weapons_locked)) {
+			Assert(icon);
+			bool done = false;
 
-				if ( class_select >= 0 ) {	// only ship select
-					if ( Carried_ss_icon.from_slot == slot_index ) {
-						if ( ss_carried_icon_moved() ) {
-							bitmap_to_draw = Wing_slot_empty_bitmap;
-						} else {
-							bitmap_to_draw = -1;
-						}
-						break;
+			if ( class_select >= 0 ) {	// only ship select
+				if ( Carried_ss_icon.from_slot == slot_index ) {
+					if ( ss_carried_icon_moved() ) {
+						bitmap_to_draw = Wing_slot_empty_bitmap;
+					} else {
+						bitmap_to_draw = -1;
 					}
+					done = true;
 				}
+			}
 
-				if ( ws->status & WING_SLOT_LOCKED ) {
```

Comment:
Removing this whole block is surely incorrect and probably related to the visual glitches, but I haven't yet figured out what to do with this.

Reply:
Looks like the only case where a slot could be simultaneously locked and something else is in multiplayer.

Since there are already plans to overhaul this code for multi, probably best not to go the route this PR is taking. Oh, well.